src/auth/controllers/auth.controller.ts

```
@@ -0,0 +1,105 @@
+    const password = request.password;
+    const user = await this.authService.validateUser(email, password);
+    if (user) {
+      const tokens = await this.authService.generateTokens(user);
+
+      AuthController.setAuthCookies(res, tokens);
+
+      return user;
+    }
+
+    // TODO: log and return a standardized error response
+    res.status(HttpStatus.UNAUTHORIZED).send({
+      message: 'Invalid credentials',
+    });
+  }
+
+  @Post('refresh')
+  @UseGuards(JwtRefreshAuthGuard)
+  async refresh(@Res({ passthrough: true }) res) {
+    const refreshToken = res.cookies[REFRESH_TOKEN_COOKIE_NAME];
+    const tokens = await this.authService.refreshTokens(refreshToken, false);
+    AuthController.setAuthCookies(res, tokens);
+  }
+
+  @Post('register')
+  async register(
+    @Body() request: CreateUserDto,
+    @Res({ passthrough: true }) res,
+  ) {
+    const name = request.name;
+    const email = request.email;
+    const password = request.password;
+    const existingUser = await this.authService.findUserByEmail(email);
+    if (existingUser) {
+      // TODO: log and return a standardized error response
+      res.status(HttpStatus.BAD_REQUEST).send({
+        message: 'Email already in use',
+      });
+      return;
+    }
+
+    const user = await this.authService.createUser(name, email, password);
+    if (user) {
+      const tokens = await this.authService.generateTokens(user);
+      AuthController.setAuthCookies(res, tokens);
+      return user;
+    }
+    // TODO: log and return a standardized error response
+    res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
+      message: 'Failed to create user',
+    });
+  }
+
+  @Post('logout')
+  async logout(@Res({ passthrough: true }) res) {
+    res.clearCookie(ACCESS_TOKEN_COOKIE_NAME);
+    res.clearCookie(REFRESH_TOKEN_COOKIE_NAME);
+    return {};
+  }
+
+  @Get('me')
+  @UseGuards(JwtAuthGuard)
+  async me(@CurrentUser() user: UserInfo) {
+    return user;
```

Is it a valid case that the user might not be found?

---

In this case, the `JwtAuthGuard` works in a way that it queries the DB for the user. If the user exists, it attaches it to the request object. If not, it halts and does not continue. The `@CurrentUser` decorator just extracts the `user` property from the request object; if `JwtAuthGuard` is used, the `@CurrentUser` will always have a value.
In other cases, if the `@CurrentUser` is used without using a guard, the user property will be undefined. So, it only makes sense to use it with the guard.

---

So, do we need to throw an error or something clarifying that the user is not found?

---

The `JwtAuthGuard` responds with 401 in case the user is not found, otherwise, we're 100% sure the user property is the currently logged-in user.

If you mean by using the `@CurrentUser` without the guard, there is no use-case for it without the guard. That's a guaranteed undefined value because the guard is the only way to set the user property.